app/views/posts/show.html.erb

```diff
@@ -0,0 +1,17 @@
+<div>
+  <h3><%= @post.title %></h3>
+  <p><%= @post.body %></p>
+</div>
+
+<% if @post.user_id == current_user.id  %>
+  <%= link_to 'Edit Post', edit_post_path(@post) %>
+  <%= link_to 'Delete Post', post_path(@post), method: :delete %>
```

Comment:
This is fine. The method: :delete ends up adding a special attribute to the link. Then rails's javascript library intercepts the click (which would be a get) and changes the request to be a delete (strictly it's a post with a _method parameter as browser's don't usually send deletes.)